src/dmd/cppmanglewin.d

```diff
@@ -669,149 +669,241 @@ private:
         buf.writeByte(cv_mod);
     }
 
-    void mangleName(Dsymbol sym, bool dont_use_back_reference = false)
+    // Return true if we are finished mangling this symbol
+    bool mangleSpecialName(Dsymbol sym)
     {
-        //printf("mangleName('%s')\n", sym.toChars());
-        const(char)* name = null;
-        bool is_dmc_template = false;
+        const(char)* mangle;
         if (sym.isCtorDeclaration())
+            mangle = "?0";
+        else if (sym.isPrimaryDtor())
+            mangle = "?1";
+        else if (!sym.ident)
+            return false;
+        else if (sym.ident == Id.assign)
+            mangle = "?4";
+        else if (sym.ident == Id.eq)
+            mangle = "?8";
+        else if (sym.ident == Id.index)
+            mangle = "?A";
+        else if (sym.ident == Id.call)
+            mangle = "?R";
+        else if (sym.ident == Id.cppdtor)
+            mangle = "?_G";
+        
+        buf.writestring(mangle);
+        return true;
+    }
+    
+    // Return true if we are finished mangling this symbol
+    bool mangleOperator(TemplateInstance ti, CppOperator whichOp, ref const(char)* symName, ref int firstTemplateArg)
```

Comment:
firstTemplateArg should be `out`, not `ref` (`symName` too?)

Reply:
Symname should actually be an `out bool` since its only used to see if it changed. `firstTemplateArg` is cleaner as a `ref` since an `out` must be assigned.